applications/nrf5340_audio/src/bluetooth/bt_content_control/content/bt_media_control.c

```diff
@@ -184,101 +236,172 @@ static void mcs_local_player_instance_cb(struct media_player *player, int err)
+ *
+ * param	cmd	Command to send
+ *
+ * @return	0 for success, error otherwise.
+ */
+static int play_pause(struct mpl_cmd *cmd)
+{
+	int ret = 0;
+
+	if (IS_ENABLED(CONFIG_BT_MCS)) {
+		ret = media_proxy_ctrl_send_command(local_player, cmd);
+	} else if (IS_ENABLED(CONFIG_BT_MCC)) {
+		/* Send cmd to all peers connected and has finished discovery */
+		for (uint8_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+			if (mcc_peer[i].mcp_mcs_disc_status == FINISHED) {
+				if (mcc_peer[i].conn != NULL) {
+					ret = bt_mcc_send_cmd(mcc_peer[i].conn, cmd);
+					if (ret) {
+						LOG_WRN("Failed to send play/pause command: %d",
+							ret);
+						return ret;
+					}
+				}
+			} else {
+				LOG_WRN("MCS discovery has not finished: %d",
+					mcc_peer[i].mcp_mcs_disc_status);
+				return -EBUSY;
+			}
+		}
+	}
+
+	if (ret) {
+		LOG_WRN("Failed to send play/pause command: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int bt_media_control_discover(struct bt_conn *conn)
 {
 	if (!IS_ENABLED(CONFIG_BT_MCC)) {
 		LOG_ERR("MCC not enabled");
 		return -ECANCELED;
 	}
 
 	int ret;
-	uint8_t idx = bt_conn_index(conn);
+	int idx = mcc_peer_index_get(conn);
 
-	if (mcp_mcs_disc_status[idx] == FINISHED || mcp_mcs_disc_status[idx] == IN_PROGRESS) {
+	if (idx == -ESRCH) {
+		idx = mcc_peer_index_free_get();
+		if (idx < 0) {
+			LOG_WRN("Error getting free index");
+			return idx;
+		}
+
+		mcc_peer[idx].conn = conn;
+	}
+
+	if (mcc_peer[idx].mcp_mcs_disc_status == FINISHED ||
+	    mcc_peer[idx].mcp_mcs_disc_status == IN_PROGRESS) {
 		return -EALREADY;
 	}
 
-	mcp_mcs_disc_status[idx] = IN_PROGRESS;
+	mcc_peer[idx].mcp_mcs_disc_status = IN_PROGRESS;
 	ret = bt_mcc_discover_mcs(conn, true);
 	if (ret) {
-		mcp_mcs_disc_status[idx] = IDLE;
+		mcc_peer[idx].mcp_mcs_disc_status = IDLE;
 		return ret;
 	}
 
 	return 0;
 }
 
-int ble_mcs_state_update(struct bt_conn *conn)
+int bt_media_control_state_update(struct bt_conn *conn)
 {
 	if (!IS_ENABLED(CONFIG_BT_MCC)) {
 		LOG_ERR("MCC not enabled");
 		return -ECANCELED;
 	}
 
-	uint8_t idx = bt_conn_index(conn);
+	int idx = mcc_peer_index_get(conn);
+
+	if (idx < 0) {
+		return idx;
+	}
 
-	if (mcp_mcs_disc_status[idx] != FINISHED) {
+	if (mcc_peer[idx].mcp_mcs_disc_status != FINISHED) {
 		LOG_WRN("MCS discovery has not finished");
 		return -EBUSY;
 	}
 
 	return bt_mcc_read_media_state(conn);
 }
 
-int ble_mcs_play_pause(struct bt_conn *conn)
+int bt_media_control_play(void)
 {
-	int ret = 0;
+	int ret;
 	struct mpl_cmd cmd;
 
-	if (conn != NULL) {
-		uint8_t idx = bt_conn_index(conn);
-
-		if (mcp_mcs_disc_status[idx] != FINISHED) {
-			LOG_WRN("MCS discovery has not finished");
-			return -EBUSY;
-		}
+	if (media_player_state != BT_MCS_MEDIA_STATE_PLAYING &&
+	    media_player_state != BT_MCS_MEDIA_STATE_PAUSED) {
+		LOG_ERR("Invalid state: %d", media_player_state);
+		return -ECANCELED;
 	}
 
 	if (media_player_state == BT_MCS_MEDIA_STATE_PLAYING) {
```

Comment:
Hmm now we check for both media_player_state != BT_MCS_MEDIA_STATE_PLAYING (above) and media_player_state == BT_MCS_MEDIA_STATE_PLAYING

Reply:
Same for bt_media_control_pause() below

Reply:
First we check that the state is valid and then we check for the specific value of the state.